### PR TITLE
ram: added error messages

### PR DESCRIPTION
**Pull request by adrianherrera**
Hey Vitaly,

I kept getting aborts and core dumps with no explanation why. I debugged with gdb and found the problem, so added some error messages.

Please let me know if the text is (not) appropriate!

### CODE REVIEW

**Comment by vitalych** [comment]
Can you explain under what circumstances you get these messages?

**Comment by adrianherrera** [comment]
The error occurred when I was using a 32-bit image with `qemu-system-i386` and used > 4GB RAM.

**Comment by vitalych** [comment]
I see, that makes sense. You can't have that much RAM in a 32-bit guest. It seems to be a legitimate case, so aborts() should be removed and use exit(-1) instead. How does vanilla qemu handle this case?

**Comment by adrianherrera** [comment]
Interestingly, vanilla QEMU boots just fine. The VM just doesn't see anything greater than 4GB. It is only when I LD_PRELOAD libs2e that the abort occurs

File: src/exec-ram.c [hunk 1]
```diff
@@ -137,10 +137,12 @@ static ram_addr_t find_ram_offset(ram_addr_t size) {
     }
 
     if (best_size == UINT_MAX) {
```
Comment (vitalych) [inline src/exec-ram.c]: Perhaps UINT_MAX check is not appropriate in this case. It can't go above 4gb.

Comment (vitalych) [reply src/exec-ram.c]: need to replace "unsigned" as well.

Comment (adrianherrera) [reply src/exec-ram.c]: Vanilla QEMU only has this check (in the same function):

```C
    if (offset == RAM_ADDR_MAX) {
        fprintf(stderr, "Failed to find gap of requested size: %" PRIu64 "\n",
                (uint64_t)size);
        abort();
    }
```

File: src/exec-ram.c [hunk 2]
```diff
@@ -137,10 +137,12 @@ static ram_addr_t find_ram_offset(ram_addr_t size) {
...
+        fprintf(stderr, "libcpu: ram size too large\n");
         abort();
     }
 
     if (best_offset == RAM_ADDR_MAX) {
+        fprintf(stderr, "libcpu: ram size too large\n");
```
Comment (vitalych) [inline src/exec-ram.c]: More precisely this error means that there is no free slot available in the list of ram blocks.

Comment (adrianherrera) [reply src/exec-ram.c]: Yes, see my previous comment :)

File: src/exec-ram.c [hunk 3]
```diff
@@ -124,7 +124,7 @@ static ram_addr_t find_ram_offset(ram_addr_t size) {
         }
 
         ram_addr_t next_gap = b->offset + b->length;
-        unsigned next_size = last ? UINT_MAX - next_gap : (b + 1)->offset - next_gap;
+        ram_addr_t next_size = last ? UINT_MAX - next_gap : (b + 1)->offset - next_gap;
```
Comment (vitalych) [inline src/exec-ram.c]: UINT_MAX is wrong here

Comment (adrianherrera) [reply src/exec-ram.c]: okay, what about now.

**Review by vitalych — APPROVED** [review]

**Pull request merged**